### PR TITLE
Reorg the book

### PR DESCRIPTION
**Pull request by garious**
#### Problem

The book primarily describes how the codebase works. The rest of the content was bolted onto the end.

#### Summary of Changes

Another attempt to reorg the book.  First document how to use the cluster, then how to develop apps, then participating in the cluster, then how the cluster works, and then finally, how to develop the validator code.

Also, swap Accepted and Implemented design proposals.

### CODE REVIEW

File: book/src/SUMMARY.md [hunk 1]
```diff
@@ -1,11 +1,30 @@
 # Table of contents
 
-* [The Solana Codebase](introduction.md)
-* [Building from Source](getting-started/README.md)
-  * [Example Client: Web Wallet](getting-started/webwallet.md)
-* [Programming Model](programs/README.md)
-  * [Example: Tic-Tac-Toe](programs/tictactoe.md)
-  * [Drones](programs/drones.md)
+* [Introduction](introduction.md)
+* [Using Solana](using-solana.md)
```
Comment (jstarry) [inline book/src/SUMMARY.md]: Not a fan of this name (too general) and the content in this section is pretty specialized (offline signing and paper wallet). How about naming it something more specific to the CLI and moving it between "Running an Archiver" and "A Solana Cluster"? Maybe "Using Solana's CLI"?

File: book/src/SUMMARY.md [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+    * [Paper Wallet Usage](paper-wallet/usage.md)
+  * [Offline Signing](offline-signing/README.md)
+    * [Durable Transaction Nonces](offline-signing/durable-nonce.md)
+* [Developing Applications](apps/README.md)
+  * [Example: Web Wallet](apps/webwallet.md)
+  * [Example: Tic-Tac-Toe](apps/tictactoe.md)
+  * [Drones](apps/drones.md)
+  * [Anatomy of a Transaction](transaction.md)
+  * [JSON RPC API](apps/jsonrpc-api.md)
+  * [JavaScript API](apps/javascript-api.md)
```
Comment (jstarry) [inline book/src/SUMMARY.md]: Big fan of this grouping for developers. But I think "JSON RPC API" section is also relevant to validators. Maybe we could add a link to it in the "Running a Validator" section as well?

File: book/src/SUMMARY.md [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+  * [Validator Requirements](running-validator/validator-reqs.md)
+  * [Choosing a Testnet](running-validator/validator-testnet.md)
+  * [Installing the Validator Software](running-validator/validator-software.md)
+  * [Starting a Validator](running-validator/validator-start.md)
+  * [Staking](running-validator/validator-stake.md)
+  * [Monitoring a Validator](running-validator/validator-monitor.md)
+  * [Publishing Validator Info](running-validator/validator-info.md)
+  * [Troubleshooting](running-validator/validator-troubleshoot.md)
+* [Running an Archiver](running-archiver.md)
 * [A Solana Cluster](cluster/README.md)
```
Comment (jstarry) [inline book/src/SUMMARY.md]: Name isn't too obvious.. How about "Understand Solana's Architecture" to match the "verb noun" naming convention we are using now?

File: book/src/SUMMARY.md (hunk 1 above)
Comment (garious) [reply book/src/SUMMARY.md]: So I'm attempting to funnel people from the most basic usage down to gaining enough expertise to make contributions to the source base.  If we move that section down, the reader jumps straight to application development, which is a smaller audience than those wanting to just transfer tokens.

Good point though, that everything in that section is command-line. I'll rename to something more specific.

File: book/src/SUMMARY.md (hunk 2 above)
Comment (garious) [reply book/src/SUMMARY.md]: To my knowledge, validators primarily use the CLI and would only use those curl commands if the CLI was missing the command.

Comment (jstarry) [reply book/src/SUMMARY.md]: Ah true, I've observed that as well. Never mind!

File: book/src/SUMMARY.md (hunk 3 above)
Comment (garious) [reply book/src/SUMMARY.md]: Fixed in bb840d4

File: book/src/SUMMARY.md (hunk 1 above)
Comment (garious) [reply book/src/SUMMARY.md]: Fixed in 8ea99a9

**Pull request merged**